scripts/core/editor3/components/toolbar/link-button/LinkPopover.jsx

```diff
@@ -0,0 +1,79 @@
+import React, {Component} from 'react';
+import {getVisibleSelectionRect} from 'draft-js';
+import * as common from '../../common';
+import $ from 'jquery';
+
+/**
+ * @ngdoc React
+ * @module superdesk.core.editor3
+ * @name LinkPopover
+ * @param {Object} editorState the state of the editor
+ * @param {Function} onChange on change function to be called when the editor
+ * state changes
+ * @description This component holds the link editing popover displayed when clicking
+ * a link in the editor and is part of LinkControl.
+ */
+export default class LinkPopover extends Component {
+    constructor(props) {
+        super(props);
+
+        this.position = null;
+
+        this.onEdit = this.onEdit.bind(this);
+        this.updatePosition = this.updatePosition.bind(this);
+    }
+
+    /**
+     * @ngdoc method
+     * @name LinkPopover#onEdit
+     * @description Called when the edit action is clicked in the popover.
+     */
+    onEdit() {
+        const {editorState} = this.props;
+        const url = common.getSelectedEntityData(editorState).url;
+
+        this.props.onEdit(url);
+    }
+
+    /**
+     * @ngdoc method
+     * @name LinkPopover#updatePosition
+     * @description Computes the position of the popover relative to the editor root.
+     */
+    updatePosition() {
+        const rect = getVisibleSelectionRect(window);
+
+        if (!rect) {
+            return;
+        }
+
+        const rootNode = $('.Editor3-root');
```

Comment:
Yes, that would be a more accurate solution. This can prove to be erroneous in cases where there are multiple editors on display (I'm not sure if we have any now).

I will try to see if I can improve this...